app/songflong/search.py

```diff
@@ -6,6 +6,10 @@
 
 import json
 
+# This code works but in reality, I don't really know why its this complex.
+# Like, why do we have a Song object? Makes no sense; especially since it requires
+# a custom serializer.
+# And what's up with the json.loads(json.dumps())? Can we fix that please?
```

Comment:
Look at my branch, I refreshed this whole file. I think a copy paste would work well (checking carefully input/output is consistent, since I may have changed that a bit)